test/sql/table_function/icu_range_timestamptz.test

```diff
@@ -0,0 +1,144 @@
+1991-07-01 00:00:00-07
+
+query I
+SELECT d FROM generate_series(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '1991-06-01 00:00:00-07', -INTERVAL '1 MONTH') tbl(d)
+----
+1992-01-01 00:00:00-08
+1991-12-01 00:00:00-08
+1991-11-01 00:00:00-08
+1991-10-01 00:00:00-07
+1991-09-01 00:00:00-07
+1991-08-01 00:00:00-07
+1991-07-01 00:00:00-07
+1991-06-01 00:00:00-07
+
+# composite interval
+query I
+SELECT d FROM range(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '1992-12-31 12:00:00-08', INTERVAL '1 MONTH 1 DAY 1 HOUR') tbl(d)
+----
+1992-01-01 00:00:00-08
+1992-02-02 01:00:00-08
+1992-03-03 02:00:00-08
+1992-04-04 03:00:00-08
+1992-05-05 04:00:00-07
+1992-06-06 05:00:00-07
+1992-07-07 06:00:00-07
+1992-08-08 07:00:00-07
+1992-09-09 08:00:00-07
+1992-10-10 09:00:00-07
+1992-11-11 10:00:00-08
+1992-12-12 11:00:00-08
+
+# DST boundaries
+query I
+SELECT d FROM generate_series(TIMESTAMPTZ '1992-04-05 00:00:00-08', TIMESTAMPTZ '1992-04-05 12:00:00-07', INTERVAL '1 HOUR') tbl(d)
+----
+1992-04-05 00:00:00-08
+1992-04-05 01:00:00-08
+1992-04-05 03:00:00-07
+1992-04-05 04:00:00-07
+1992-04-05 05:00:00-07
+1992-04-05 06:00:00-07
+1992-04-05 07:00:00-07
+1992-04-05 08:00:00-07
+1992-04-05 09:00:00-07
+1992-04-05 10:00:00-07
+1992-04-05 11:00:00-07
+1992-04-05 12:00:00-07
+
+query I
+SELECT d FROM generate_series(TIMESTAMPTZ '1992-10-25 00:00:00-07', TIMESTAMPTZ '1992-10-25 12:00:00-08', INTERVAL '1 HOUR') tbl(d)
+----
+1992-10-25 00:00:00-07
+1992-10-25 01:00:00-07
+1992-10-25 01:00:00-08
+1992-10-25 02:00:00-08
+1992-10-25 03:00:00-08
+1992-10-25 04:00:00-08
+1992-10-25 05:00:00-08
+1992-10-25 06:00:00-08
+1992-10-25 07:00:00-08
+1992-10-25 08:00:00-08
+1992-10-25 09:00:00-08
+1992-10-25 10:00:00-08
+1992-10-25 11:00:00-08
+1992-10-25 12:00:00-08
+
+
+# large result
+query I
+SELECT COUNT(*) FROM range(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '2020-01-01 00:00:00-08', INTERVAL '1 DAY') tbl(d)
+----
+10227
+
+query I
+SELECT COUNT(*) FROM generate_series(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '2020-01-01 00:00:00-08', INTERVAL '1 DAY') tbl(d)
+----
+10228
+
+# zero interval not supported
+statement error
+SELECT d FROM range(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '1992-12-31 12:00:00-08', INTERVAL '0 MONTH') tbl(d)
+
+# start is smaller than end but we have a negative interval
+statement error
+SELECT d FROM range(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '1992-12-31 12:00:00-08', INTERVAL '1 MONTH ago') tbl(d)
+
+# start is bigger than end but we have a positive interval
+statement error
+SELECT d FROM range(TIMESTAMPTZ '1993-01-01 00:00:00-08', TIMESTAMPTZ '1992-01-01 00:00:00-08', INTERVAL '1 MONTH') tbl(d)
+
+# composite interval with negative types not supported
+statement error
+SELECT d FROM range(TIMESTAMPTZ '1992-01-01 00:00:00-08', TIMESTAMPTZ '1992-12-31 12:00:00-08', INTERVAL '1 MONTH' - INTERVAL '1 HOUR') tbl(d)
+
+# Infinities will overflow or cause infinite loops (PG behaviour!) so we ban them
```

Comment:
Heh PG acknowledged the bug for the last release and fixed it. I have a medallion to prove it 😄

Reply:
That's great!
Finding a bug in edge cases seems a good way for me to get the PG medallion 🤔